src/utils/work/waitForWorkRequest.js

```diff
@@ -154,6 +114,7 @@ const dispatchWorkRequest = async (
 
   const responsePromise = new Promise((resolve, reject) => {
     io.on(`WorkResponse-${ETag}`, async (res) => {
```

Comment:
I see that places where we were emitting `WorkResponse` are deleted in the api, but the UI still listens on `WorkResponse`. Is this `responsePromise` still used? If so, how does it work?

Reply:
I see this line in the worker: `io.Emit(f'WorkResponse-{self.request["ETag"]}', self._construct_response_msg(socket_data))`. So the worker must be submitting the response. Is this right?

Reply:
cleared this in slack -- that is correct, the worker returns the response, which is either data (if the response is small enough) or signedURL otherwise

Reply:
if the API returns a signed URL (meaning data is already in S3) it will not dispatch the request to the worker, so the worker will not return anything to the uI